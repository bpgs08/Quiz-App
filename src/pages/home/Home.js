import React from "react";
import styled from "styled-components";
import Headings from "../../components/Headings";
import { space, media } from "../../utils/theme";
import { Link } from "react-router-dom";

let Home = () => {
  return (
    <HomeContainer>
      <StyledHeading type={5} bold={true} align="center">
        Welcome to the Trivia Challenge!
      </StyledHeading>
      <Paragraph>
        You will be presented with 10 True or False questions.
      </Paragraph>
      <Paragraph>Can you score 100%</Paragraph>

      <BeginButton to={`/quiz`}>BEGIN</BeginButton>
    </HomeContainer>
  );
};

const StyledHeading = styled(Headings)`
  margin: ${space[3]} auto 0 auto;
  max-width: 180px;
  ${media.tablet`
    max-width: 225px;
  `}
`;

const HomeContainer = styled.div`
  margin: ${space[0]} auto;
  padding: ${space[0]} ${space[3]};
  max-width: 300px;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  ${media.tablet`
    margin: ${space[0]} auto;
    max-width: 400px;
  `}
`;

const Paragraph = styled.p`
  margin: auto auto 0 auto;
  max-width: 180px;
  text-align: center;
`;

const BeginButton = styled(Link)`
  margin: auto auto ${space[5]} auto;
  cursor: pointer;
  text-decoration: none;
  text-align: center;
  color: black;
`;

export default Home;
