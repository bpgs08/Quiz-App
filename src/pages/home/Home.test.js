import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Home from "./Home";

test("it works", () => {
  const { getByText, getByLabelText } = render(
    <Router>
      <Home />
    </Router>
  );

  getByText("Welcome to the Trivia Challenge!");
  getByText("You will be presented with 10 True or False questions.");
  getByText("Can you score 100%");
  getByText("BEGIN");
});

test("should take a snapshot", () => {
  const { asFragment } = render(
    <Router>
      <Home />
    </Router>
  );

  expect(
    asFragment(
      <Router>
        <Home />
      </Router>
    )
  ).toMatchSnapshot();
});

// test("add to list", async () => {
//   const { getByText, getByLabelText } = render(
//     <Router>
//       <Home />
//     </Router>
//   );

//   getByText("Welcome to the Trivia Challenge!");
//   getByText("You will be presented with 10 True or False questions.");
//   getByText("Can you score 100%");
//   getByText("BEGIN");

//   const button = getByText("BEGIN");
//   userEvent.click(button);
//   await waitFor(() => getByText("BEGINs"));
// });
