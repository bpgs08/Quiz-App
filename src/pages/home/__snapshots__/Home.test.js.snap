// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-iCoHVE gZdfGi"
  >
    <h5
      class="sc-eCApGN gaEGIs sc-gKAblj jWetxF"
      data-js-type-5="true"
    >
      Welcome to the Trivia Challenge!
    </h5>
    <p
      class="sc-fujyUd bXTDXY"
    >
      You will be presented with 10 True or False questions.
    </p>
    <p
      class="sc-fujyUd bXTDXY"
    >
      Can you score 100%
    </p>
    <a
      class="sc-pNWxx hXXOLl"
      href="/quiz"
    >
      BEGIN
    </a>
  </div>
</DocumentFragment>
`;
